pkg/content_download.go

```diff
@@ -0,0 +1,84 @@
+package pkg
+
+import (
+	"github.com/wttech/aemc/pkg/common/filex"
+	"github.com/wttech/aemc/pkg/common/pathx"
+	"github.com/wttech/aemc/pkg/content"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const (
+	FilterXml = "filter.xml"
+)
+
+type Downloader struct {
+	config *content.Opts
+}
+
+func NewDownloader(config *content.Opts) *Downloader {
+	return &Downloader{
+		config: config,
+	}
+}
+
+func (c Downloader) DownloadPackage(packageManager *PackageManager, pid string, roots []string, filter string) (string, error) {
+	if pid == "" {
+		pid = "my_packages:aemc_content:" + time.Now().Format("2006.102.304") + "-SNAPSHOT"
+	}
+	remotePath, err := packageManager.Create(pid, roots, filter)
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		_ = packageManager.Delete(remotePath)
+	}()
+	if err := packageManager.Build(remotePath); err != nil {
+		return "", err
+	}
+	tmpResultFile := filepath.Join(c.config.BaseOpts.TmpDir, filepath.Base(remotePath))
+	if err := packageManager.Download(remotePath, tmpResultFile); err != nil {
+		return "", err
+	}
+	return tmpResultFile, nil
+}
+
+func (c Downloader) DownloadContent(packageManager *PackageManager, pid string, root string, roots []string, filter string, clean bool, unpack bool) error {
```

Comment:
this method should be in another place, rather unpacking should be disconnected from downloading